modules/phase_field/src/materials/GammaAnisoGauss.C

```diff
@@ -0,0 +1,582 @@
+    _m_EN(declareProperty<Real>(getParam<MaterialPropertyName>("m_EN_name"))),
+    _kappa(declareADProperty<Real>(getParam<MaterialPropertyName>("kappa_name"))),
+    _kappa_EN(declareProperty<Real>(getParam<MaterialPropertyName>("kappa_EN_name"))),
+    _L(declareADProperty<Real>(getParam<MaterialPropertyName>("L_name"))),
+    _L_EN(declareProperty<Real>(getParam<MaterialPropertyName>("L_EN_name"))),
+    _lgb(declareADProperty<Real>(getParam<MaterialPropertyName>("lgb_name"))),
+    _lgb_EN(declareProperty<Real>(getParam<MaterialPropertyName>("lgb_EN_name"))),
+    _sigma(declareADProperty<Real>(getParam<MaterialPropertyName>("sigma_name"))),
+    _sigmaORIUNIT(declareADProperty<Real>(getParam<MaterialPropertyName>("sigmaORIUNIT_name"))),
+    _qwg(declareADProperty<Real>(getParam<MaterialPropertyName>("qwg_name"))),
+    _qxg(declareADProperty<Real>(getParam<MaterialPropertyName>("qxg_name"))),
+    _qyg(declareADProperty<Real>(getParam<MaterialPropertyName>("qyg_name"))),
+    _qzg(declareADProperty<Real>(getParam<MaterialPropertyName>("qzg_name"))),
+    _sigmaBASE(getParam<Real>("sigmaBASE")),
+    _GgammaBASE(getParam<Real>("GgammaBASE")),
+    _gammaBASE(getParam<Real>("gammaBASE")),
+    _f0gammaBASE(getParam<Real>("f0gammaBASE")),
+    _lgbBASE_minimum(getParam<Real>("lgbBASE_minimum")),
+    _alphaswitch(getParam<Real>("alphaswitch")),
+    _betaswitch(getParam<Real>("betaswitch")),
+    _libnum(getParam<int>("libnum")),
+    _amplitudeScale(getParam<Real>("amplitudeScale")),
+    _sharpness(getParam<Real>("sharpness")),
+    _Gaussian_Tolerance(getParam<Real>("Gaussian_Tolerance")),
+    _Library_file_name(getParam<FileName>("Library_file_name")),
+    _Quaternion_file_name(getParam<FileName>("Quaternion_file_name")),
+    _grain_tracker(getUserObject<GrainTrackerInterface>("grain_tracker")),
+    _length_scale(getParam<Real>("length_scale")),
+    _time_scale(getParam<Real>("time_scale")),
+    _JtoeV(getParam<Real>("JtoeV")),
+    _GBMobility(getParam<Real>("GBMobility")),
+    _Mob(getParam<Real>("Mob")),
+    _Q(getParam<Real>("Q")),
+    _T(getParam<Real>("T")),
+    _kb(getParam<Real>("kb")),
+    _BoundaryNormal(getParam<Real>("BoundaryNormal")),
+    _op_num(coupledComponents("v")),
+    _vals(adCoupledValues("v")),
+    _grad_vals(adCoupledGradients("v"))
+{
+  if (_op_num == 0)
+    paramError("op_num", "op_num must be greater than 0");
+
+  if (_GBMobility == 1 && _Mob == 0 && _Q == 0)
+    mooseError("Enter GBMobility or Mob and Q");
+
+  // Read data files during initialization
+  readQuaternionFile();
+  readLibraryFile();
+}
+
+void
+GammaAnisoGauss::readQuaternionFile()
+{
+  std::ifstream File1(_Quaternion_file_name.c_str());
+  if (!File1.is_open())
+    mooseError("Unable to open quaternion file");
+
+  double QuatW, QuatX, QuatY, QuatZ;
+  while (File1 >> QuatW >> QuatX >> QuatY >> QuatZ)
+  {
+    _qwR.push_back(QuatW);
+    _qxR.push_back(QuatX);
+    _qyR.push_back(QuatY);
+    _qzR.push_back(QuatZ);
+  }
```

Comment:
are there some checks we should be doing on these variables? like normalization?

Reply:
See updated files

I have made it so that they are normalized in the code.